BUILD.md

```diff
@@ -0,0 +1,472 @@
+  - [`public-api`](controller/api/public): Accepts requests from API
+    clients such as `cli` and `web`, provides access to and control of the
+    conduit service mesh.
+  - [`tap`](controller/tap): Provides a live pipeline of requests.
+  - [`telemetry`](controller/telemetry): Collects and aggregates
+    metrics from `proxy` componenets.
+- [`proxy-init`](proxy-init): Adds a Kubernetes pod to join the Conduit
+  Service Mesh.
+- [`web`](web): Provides a UI dashboard to view and drive the control plane.
+
+## Data Plane (Rust)
+
+  - [`proxy`](proxy): High-performance data plane, injected as a sidecar with
+    every service.
+  - [`tower-grpc`](tower-grpc): A client and server gRPC implementation based on
+    Tower.
+  - [`tower-grpc-examples`](tower-grpc-examples): Demonstrates how to use Tower
+    gRPC clients and servers with code generation.
+  - [`tower-h2`](tower-h2): Tower `Service` abstractions for HTTP/2 and Rust.
+  - [`tower-router`](tower-router): A Tower middleware that routes requests to
+    one of a set of inner services using a request predicate.
+
+# Components
+
+![Conduit Components](https://g.gravizo.com/source/svg/conduit_components?https%3A%2F%2Fraw.githubusercontent.com%2Frunconduit%2Fconduit%2Fsiggy%2Fbuild%2FBUILD.md)
+
+<details>
+<summary></summary>
+conduit_components
+  digraph G {
+    rankdir=LR;
+
+    node [style=filled, shape=rect];
+
+    "cli" [color=lightblue];
+    "destination" [color=lightblue];
+    "proxy-api" [color=lightblue];
+    "public-api" [color=lightblue];
+    "tap" [color=lightblue];
+    "telemetry" [color=lightblue];
+    "web" [color=lightblue];
+
+    "proxy" [color=orange];
+
+    "cli" -> "public-api";
+    "web" -> "public-api";
+
+    "destination" -> "kubernetes";
+
+    "proxy" -> "proxy-api";
+
+    "proxy-api" -> "destination";
+    "proxy-api" -> "telemetry";
+
+    "public-api" -> "tap";
+    "public-api" -> "telemetry";
+
+    "tap" -> "proxy";
+
+    "telemetry" -> "prometheus";
+  }
+conduit_components
+</details>
+
+# Development configurations
+
+Depending on use case, there are several configurations with which to develop
+and run Conduit:
+
+- [Minikube](#minikube): Comprehensive config, most closely matches release.
+- [Docker Compose](#docker-compose): Development of the Go components.
+- [Web](#web): Development of the Conduit Dashboard.
+- [Proxy](#proxy): Standalone development of the Rust `proxy`.
+
+## Minikube
+
+This configuration builds all Conduit components in Docker images, and deploys
+them onto Minikube. This configuration also builds and installs a `conduit`
+executable onto the local system. This setup most closely parallels our
+recommended production installation, documented at
+https://conduit.io/getting-started/.
+
+These commands assume working Go and Minikube environments.
+
+```bash
+# ensure all go dependencies are in vendor
+dep ensure && dep prune
+
+# build and install conduit cli locally
+go build -o $GOPATH/bin/conduit ./cli
```

Comment:
Personal preference, but I don't usually install the local cli into my global bin dir. Instead, I just run the cli locally with, for example:

```
bin/go-run cli version
```